src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c

```diff
@@ -21,77 +46,699 @@ const ActorInit En_Goroiwa_InitVars = {
+        this->actor.posRot.pos.y = nextPointY - ((this->actor.posRot.pos.y - nextPointY) * 0.3f);
+    }
+    if (this->unk_1C6 == 0 &&
+        func_80042244(globalCtx, &globalCtx->colCtx, this->actor.posRot.pos.x, this->actor.posRot.pos.z, &ySurface,
+                      &waterBox) &&
+        this->actor.posRot.pos.y <= ySurface) {
+        this->unk_1D3 |= 0x10;
+        if (ySurface < thisY) {
+            sp30.x = this->actor.posRot.pos.x;
+            sp30.y = ySurface;
+            sp30.z = this->actor.posRot.pos.z;
+            func_80A4C594(globalCtx, &sp30);
+            this->actor.velocity.y *= 0.2f;
+        }
+        if (this->actor.velocity.y < -8.0f) {
+            this->actor.velocity.y = -8.0f;
+        }
+    }
+    return false;
+}
+
+void func_80A4CED8(EnGoroiwa* this, GlobalContext* globalCtx) {
+    static Vec3f unitY = { 0.0f, 1.0f, 0.0f };
+
+    s32 pad;
+    Vec3f* temp;
+    f32 sp8C;
+    Vec3f sp80;
+    Vec3f sp74;
+    MtxF mtx;
+    Vec3f sp28;
+
+    if (this->unk_1D3 & 8) {
+        sp8C = this->unk_1BC;
+    } else {
+        this->unk_1BC = Math3D_Vec3f_DistXYZ(&this->actor.posRot.pos, &this->actor.pos4) * 0.016806724f;
+        sp8C = this->unk_1BC;
+    }
+    sp8C *= this->unk_1C0;
+    // temp assignment needed to match
+    temp = &sp80;
+    if (this->unk_1D3 & 8) {
+        func_80A4BF28(this, globalCtx, &sp28);
+        Math3D_Vec3f_Cross(&unitY, &this->actor.velocity, temp);
+    } else {
+        Math3D_Vec3f_Cross(&unitY, &this->actor.velocity, temp);
+    }
+
+    if (EnGoroiwa_Vec3fNormalize(&sp74, temp)) {
+        this->unk_1B0 = sp74;
+    } else {
+        sp74 = this->unk_1B0;
+    }
+
+    func_800D23FC(sp8C, &sp74, 0);
+    Matrix_RotateY(this->actor.shape.rot.y * (2.0f * M_PI / 65536.0f), 1);
+    Matrix_RotateX(this->actor.shape.rot.x * (2.0f * M_PI / 65536.0f), 1);
+    Matrix_RotateZ(this->actor.shape.rot.z * (2.0f * M_PI / 65536.0f), 1);
+    Matrix_Get(&mtx);
+    func_800D20CC(&mtx, &this->actor.shape.rot, 0);
+}
+
+void func_80A4D074(EnGoroiwa* this, GlobalContext* globalCtx) {
+    s16 temp_v0 = (this->actor.params >> 8) & 3;
+
+    func_80A4C134(this);
+
+    if (temp_v0 == 0 || temp_v0 == 1) {
+        if (this->currentWaypoint == 0 || this->currentWaypoint == this->endWaypoint) {
+            func_80A4C1C4(this, globalCtx, this->currentWaypoint);
+        }
+    }
+
+    func_80A4BE54(this, globalCtx);
+}
+
+void func_80A4D0FC(EnGoroiwa* this, GlobalContext* globalCtx) {
+    static f32 colliderHeightOffset[] = { 0.0f, 59.5f };
+
+    s16 angle1;
+    s16 angle2;
+    s32 pad;
+    Vec3f* thisPos = &this->actor.posRot.pos;
+    Vec3f effectPos;
+    Vec3f effectVelocity;
+    f32 temp_f24;
+    f32 temp_f22;
+    f32 temp_f20;
+    f32 temp_f20_2;
+    s16 temp_v0 = (this->actor.params >> 10) & 1;
+    s32 i;
+
+    for (i = 0, angle1 = 0; i < 16; i++, angle1 += 0x4E20) {
+        temp_f22 = Math_Sins(angle1);
+        temp_f24 = Math_Coss(angle1);
```

Comment:
can `i * 0x4E20` be used instead of angle1?

Reply:
I don't believe so. I've tried that and couldn't get it to match.